apache/ambari
# AMBARI-24516 - Default value for LDAP type

## What changes were proposed in this pull request?

- Adding query ldap type broke existing QE scripts. For backward compatibility a new cmd line option added: --ldap-use-generic-defaults - Disables ldap type query and generic defaults will be offered for properties which are not exists and not specified in the command line.
- Ask for ldap type only if any of the properties which default value is depending from ldap type is missing because it was not added during previous ldap setup or not specified in the command line
- Ask for the user credentials and queries Ambari for the existing values first. Than offer these values as defaults

## How was this patch tested?

Manual tests:

1. Install Ambari and replace the affected .py files.
2. start ambari server
3. Run ambari-server setup-ldap
4. check whether the ldap config properties are saved to the database table ambari_configuration
5. Restart ambari server
6. Run ambari-server sync-ldap
7. Log into the ambari server web UI using one of the users stored in ldap

Run through different paths of the wizard like

- choosing different type of ldaps
- select use ssl true/false
- select Bind anonymously true/false
- set ambari.ldap.advanced.collision_behavior convert/skip
- set ambari.ldap.advanced.disable_endpoint_identification true/false
